salt/client/ssh/__init__.py

```diff
@@ -911,6 +911,7 @@ def __init__(
                     'id': self.id,
                     'sock_dir': '/',
                     'fileserver_list_cache_time': 3,
+                    'pillar_compile_merge_global': True
```

Comment:
This gets set to `True` all the time and ignores _any kind_ of settings. I would be OK to set it to `True` by default, but completely ignoring the opposite value of it isn't the great idea. And maybe shorter name? like `merge_pillar`.